src/app/application.cpp

```diff
@@ -31,10 +31,6 @@
 
 #include <algorithm>
 
-#ifdef Q_OS_WIN
-#include <memory>
```

Comment:
This move doesn't comply with the guidelines.
you should make it so instead:
```c++
#include <QtGlobal>  // add this

#include <algorithm>

#ifdef Q_OS_WIN
#include <memory>
#endif
```

Reply:
Isn't `<memory>` a standard library header but `<Shellapi.h>` a system header? Thus making this fall under the:
`If there are conditionals that contain headers from several different header groups, then put them above the "qBittorrent's own headers" group.`

Reply:
Apparently I wasn't clear enough when writing that part, that part should be an exception in my original intention.
OK, don't change unless other devs agrees with me.

Reply:
Well, let's compare the following examples.
```c++
#include <std_header_1>
#ifdef COND
#include <std_header_2>
#endif

#include <sys_header_1>
#ifdef COND
#include <sys_header_2>
#endif

#include <libt_header_1>
#ifdef COND
#include <libt_header_2>
#endif

#include <qt_header_1>
#ifdef COND
#include <qt_header_2>
#endif
```
```c++
#include <std_header_1>
#include <sys_header_1>
#include <libt_header_1>
#include <qt_header_1>

#ifdef COND
#include <std_header_2>
#include <sys_header_2>
#include <libt_header_2>
#include <qt_header_2>
#endif
```

The more conditions there are, the worse the first example will be. So I vote for the second example. Conditionals should be on the first level.